packages/core/src/actions/smartAccount/signUserOperation.ts

```diff
@@ -42,12 +43,14 @@ export async function signUserOperation<
     throw new ChainNotFoundError();
   }
 
-  return await client.middleware
-    .signUserOperation(args.uoStruct, {
-      ...args,
-      account,
-      client,
-    })
-    .then(resolveProperties)
-    .then(deepHexlify);
+  const request = deepHexlify(args.uoStruct);
+  if (!isValidRequest(request)) {
+    throw new InvalidUserOperationError(args.uoStruct);
+  }
+
+  const entryPoint = account.getEntryPoint();
+  request.signature = await account.signUserOperationHash(
+    entryPoint.getUserOperationHash(request)
+  );
+  return request as UserOperationRequest<TEntryPointVersion>;
```

Comment:
this method shouldn't be using middleware, but instead be calling the account's signUOHash method

Reply:
`signUserOperationHash` is meant to be a pass through to the account's method. the default signing middleware just calls `signUoHash` , but custom signer middleware is used to change that logic or leverage context for changing the contents to be signed by the account. Think of it this way, if `client.signUserOperationHash` called the middleware and the middleware is meant to call `signHash`, then you have a circular dependency. The account passthrough actions are just a convenience so instead of doing `client.account.foo` you can just do `client.foo` (eg. `client.account.address` vs `client.getAddress`)

The original intent of the Signer Middleware is that it was used during the `runMiddlewareStack`. multisig expected the signing middleware to run during all calls that call the middleware stack (including `buildUserOperation`). It was meant for custom signing logic not on the account (`account.signUserOperationHash` is for custom signing logic) but for custom signing logic during sending a UO. For multisig, multiple accounts instance might need to sign a UO hash which they do without the client, but when sending a UO (or building a UO) the sending client is responsible for generating a valid UO. This is where the middleware is used

Reply:
Signer Middleware is NOT used during the `runMiddlewareStack`. middleware pipeline is not supposed to sign the user operation struct.

this signUserOperation action is a SmartAccountClientAction, not SmartAccountAction. runMiddlewareStack does NOT call the signer middleware, but instead _sendUseroperation call as a one-off signer middleware run that defaults to account signUserOperationHash (userOpSigner), but can be customized if the signer middleware is overridden.

Reply:
TLDR: 

> "client.signUserOperationHash called the middleware and the middleware is meant to call signHash, then you have a circular dependency."

--> `signUserOperationHash` is an account method, and client does not have `signUserOperationHash`, but `signUserOperation` action. The signer middleware only default have the connected account call `signUserOperationHash`.